### PR TITLE
Fix scheduled benchmarks

### LINKED ISSUE #2783: Benchmarks are failing
Some of the benchmarks are failing due to python errors. Others seem to be working but in the final comparison all say "failed". Therefore, the benchmarks are not currently being checked.

### PR DESCRIPTION
**Pull request by Saransh-cpp**
# Description

Benchmarks running here - https://github.com/Saransh-cpp/PyBaMM/actions/runs/4469585290/jobs/7851901865

Fixes #2783 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ pre-commit run` (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [ ] All tests pass: `$ python run-tests.py --all`
- [ ] The documentation builds: `$ python run-tests.py --doctest`

You can run unit and doctests together at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: .github/workflows/periodic_benchmarks.yml [hunk 1]
```diff
@@ -24,12 +24,22 @@ jobs:
...
           python-version: 3.8
-      - name: Install tox and asv
-        run: pip install -U pip "tox<4" asv
+      - name: Install Linux system dependencies
+        run: |
+          sudo apt-get update
+          sudo apt install gfortran gcc libopenblas-dev
+      - name: Install python dependencies
+        run: |
+          python -m pip install --upgrade pip wheel setuptools virtualenv asv wget cmake casadi numpy "tox<4" build
```
Comment (Saransh-cpp) [inline .github/workflows/periodic_benchmarks.yml:33]:
```suggestion
          python -m pip install --upgrade pip wheel setuptools virtualenv asv wget cmake casadi numpy
```

**Review by tinosulzer — COMMENTED** [review]
Where can we see if this is working?

**Comment by Saransh-cpp** [comment]
You can see the failing benchmarks here - https://github.com/Saransh-cpp/PyBaMM/actions/runs/4469585290/jobs/7851901865

The benchmarks are failing because of PyBaMM related errors - https://github.com/pybamm-team/PyBaMM/pull/2784#issuecomment-1476491320 (same for both the PRs)

**Comment by Saransh-cpp** [comment]
@tinosulzer I think this should still be merged as the scheduled benchmarks have a different workflow and that is also broken. However, I cannot re-open this PR for some reason (maybe because I rebased the branch while the PR was closed). I'll open a new PR for this.

**Pull request closed without merging**